### PR TITLE
feat: add configurable redirect

### LINKED ISSUE #141: Redirect to dweb.link

### PR DESCRIPTION
**Pull request by vasco-santos**
Adds configurable redirect via a secret that can easily be deployed and removed if needed. Later on, we can decide to remove the race code

Also made github actions use Node 16 because apparently it started installing Node18 now and we require Node16 https://github.com/web3-storage/reads/blob/main/package.json#L46 . In follow up PR I will also update to Node18

Closes #141

### CODE REVIEW

**Comment by vasco-santos** [comment]
@alanshaw @olizilla per what we talked about earlier today, just made sure and in grafana we track status code https://daghouse.grafana.net/d/PCqBFbzVz/w3link?orgId=1 . So, we will already be able to see the amount of traffic that is not served by Cache + Freeway and is redirected. So, we don't really need any action in this PR for metrics :)

File: packages/edge-gateway/src/gateway.js [hunk 1]
```diff
@@ -115,7 +115,16 @@ export async function gatewayGet (request, env, ctx) {
...
   }
 
-  // 3rd layer resolution - Public Gateways race
+  // 3rd layer resolution - Redirect to Public Gateway if set
+  if (env.ipfsGatewayRedirectHostname) {
+    const url = new URL(
+      `https://${cid}.ipfs.${env.ipfsGatewayRedirectHostname}${pathname}${search}`
+    )
+
+    return Response.redirect(url.toString(), 303)
```
Comment (alanshaw) [inline packages/edge-gateway/src/gateway.js:124]: > [303] indicates that the redirects don't link to the requested resource itself, but to another page (such as a confirmation page, a representation of a real-world object — see [HTTP range-14](https://en.wikipedia.org/wiki/HTTPRange-14) — or an upload-progress page). This response code is often sent back as a result of [PUT](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/PUT) or [POST](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/POST). The method used to display this redirected page is always [GET](https://developer.mozilla.org/en-US/docs/Web/HTTP/Methods/GET).

Maybe we want https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/307 instead?

**Review by alanshaw — APPROVED** [review]
Code changes LGTM, but I think the status code needs to change...

File: packages/edge-gateway/src/gateway.js (hunk 1 above)
Comment (vasco-santos) [reply packages/edge-gateway/src/gateway.js:124]: @olizilla 's proposal was 303 https://hackmd.io/@olizilla/w3s-redirect . So, would like to get input if there was something that made 303 the suggestion.

For the documentation, I agree that 307 seems more appropriate

Comment (olizilla) [reply packages/edge-gateway/src/gateway.js:124]: yep, agree, 307 sounds closer to what we want. I didn't realise that "303 See other" implied we were linking to a different resource.

**Pull request merged**